src/leiningen/bat_test.clj

```diff
@@ -43,24 +43,28 @@
       (if watch?
         `(do
            (System/setProperty "java.awt.headless" "true")
+           (metosin.bat-test.impl/run ~opts)
            (metosin.bat-test.impl/enter-key-listener ~opts)
-           @(watchtower.core/watcher
-              ~watch-directories
-              (watchtower.core/rate 100)
-              (watchtower.core/file-filter watchtower.core/ignore-dotfiles)
-              (watchtower.core/file-filter (watchtower.core/extensions :clj :cljc))
-              (watchtower.core/on-change (fn [~'_]
-                                           (println)
-                                           (try
-                                             (metosin.bat-test.impl/run ~opts)
-                                             (catch Exception e#
-                                               (println e#)))))))
+           (hawk.core/watch! [{:paths ~watch-directories
+                               :filter hawk.core/file?
+                               :context (constantly 0)
+                               :handler (fn [~'ctx ~'e]
+                                          (if (and (clojure.string/ends-with? (.getName (:file ~'e)) ".clj")
+                                                   (= :modify (:kind ~'e))
```

Comment:
Maybe no need to filter by event kind, because new file should also trigger tests?

And to match previous code, should ignore events where filename starts with dot. As some editor could create hidden temp files with same extension.

Reply:
Agreed. Done.